### PR TITLE
ISPN-8912 SoftIndexFileStore does not fail gracefully if max node sizee exceeds 32767

### PR DESCRIPTION
**Pull request by rvansa**
* validate the configuration
* use shorts where the value should be limited to short
* use i18d logging
* minor code cleanup based on IDE suggestions

https://issues.jboss.org/browse/ISPN-8912

### CODE REVIEW

File: persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/Index.java [hunk 1]
```diff
@@ -342,17 +328,18 @@ private void shutdown() throws IOException {
...
          buffer.putInt(0, freeBlocks.size());
          write(indexFile, buffer);
-         for (Map.Entry<Integer, List<IndexSpace>> entry : freeBlocks.entrySet()) {
+         for (Map.Entry<Short, List<IndexSpace>> entry : freeBlocks.entrySet()) {
             List<IndexSpace> list = entry.getValue();
             int requiredSize = 8 + list.size() * 10;
             buffer = buffer.capacity() < requiredSize ? ByteBuffer.allocate(requiredSize) : buffer;
             buffer.position(0);
             buffer.limit(requiredSize);
+            // TODO: change this to short
```
Comment (ryanemerson) [inline persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/Index.java]: Did you forget about this?

File: persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/Index.java [hunk 2]
```diff
@@ -389,7 +376,9 @@ private void loadFreeBlocks(long freeBlocksOffset) throws IOException {
             if (!read(indexFile, buffer)) {
                throw new IOException("Cannot read free blocks lists!");
             }
+            // TODO: change this to short
```
Comment (ryanemerson) [inline persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/Index.java]: ?

File: persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/IndexNode.java [hunk 3]
```diff
@@ -861,8 +841,10 @@ private static int compare(byte[] first, byte[] second) {
...
 
-   private int headerLength() {
-      return INNER_NODE_HEADER_SIZE + prefix.length;
+   private short headerLength() {
+      int headerLength = INNER_NODE_HEADER_SIZE + prefix.length;
+      assert headerLength <= Short.MAX_VALUE;
+      return (short) headerLength;
    }
 
    private int contentLength() {
```
Comment (ryanemerson) [inline persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/IndexNode.java]: Should be `short`

File: persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java [hunk 4]
```diff
@@ -241,7 +240,8 @@ public void stop() {
          indexQueue = null;
          storeQueue = null;
       } catch (InterruptedException e) {
-         throw new PersistenceException("Cannot stop cache store", e);
+         Thread.currentThread().interrupt();
```
Comment (wburns) [inline persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java]: I wouldn't normally expect to reinterpret the thread when throwing an exception that stating that we were interrupted. I find I would normally reset the status and return early or throw the exception, but not both.

File: persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/Index.java (hunk 1 above)
Comment (rvansa) [reply persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/Index.java]: No; I didn't want to change binary format in some micro - the plan is to add some version numbers both to index and to data files, and then I'll do the change to binary format as well.

File: persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java (hunk 4 above)
Comment (rvansa) [reply persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java]: OK, I have thought the best practice is to set thread status to interrupted if the thrown exception is not directly `InterruptedException`.

Comment (wburns) [reply persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/SoftIndexFileStore.java]: Yeah I was thinking it was a different exception. Seems good.

**Comment by wburns** [comment]
Integrated into master, thanks @rvansa !

**Pull request merged**